drivers/classic/src/calc_air_temperature.c

```diff
@@ -176,48 +174,56 @@ HourlyT(int     Dt,
 
 /******************************************************************************
  * @brief    This function estimates the times of minimum and maximum
- *           temperature for each day of the simulation, based on the hourly
+ *           temperature for each day of the simulation, based on the diurnal
  *           cycle of incoming solar radiation.
  *****************************************************************************/
 void
-set_max_min_hour(double *hourlyrad,
-                 int     ndays,
-                 int    *tmaxhour,
-                 int    *tminhour)
+set_max_min_sec(double *subdailyrad,
+                int     ndays,
+                int     stepsize,
+                int    *tmaxsec,
+                int    *tminsec)
 {
-    int risehour;
-    int sethour;
-    int hour;
-    int i;
+    size_t step;
+    int    risesec;
+    int    setsec;
+    int    sec;
+    int    i;
+    size_t stepsperday;
+
+    stepsperday = SEC_PER_DAY / stepsize;
```

Comment:
This is a good point.  I'm going to pass a `stepsperday` variable into the function and then calculate the `stepsize` inside the function.  

In the global_param_struct, I store these values:
```C
    double dt;                     /**< Time step in seconds */
    double snow_dt;                /**< Snow model time step in seconds */
    double runoff_dt;              /**< Runoff time step in seconds */
    double out_dt;                 /**< Output time step in seconds */
    size_t model_steps_per_day;    /**< Number of model timesteps per day */
    size_t snow_steps_per_day;     /**< Number of snow timesteps per day */
    size_t runoff_steps_per_day;   /**< Number of runoff timesteps per day */
```
but those do not necessarily correspond to the stepsize of this function.